### DISCUSSION TITLE
Question about static files

### DISCUSSION THREAD

**Discussion by daya0576** [zauberzeug/nicegui#3858] [Q&A] [answered]
### Question

I noticed that the behaviour of `Cache-Control` for static files is inconsistent:
```
# Response with Cache-Control header
- add_static_file
- libraries/components/resources

# Response without Cache-Control header 
- add_static_files
- static
```

Is this by design or incorrect? Thanks in advance.

**Comment by python-and-novella** [comment]
Where do you get this? Document? Or doc string?
actually, add_static_files bases on starlette's Staticfiles, you can look it up on internet.

**Reply by daya0576** [reply]
For example, when using `add_static_files` to add a directory of static files, the response header does not contain the `Cache-Control` to cache static files locally.
- https://nicegui.io/documentation/section_pages_routing#add_a_directory_of_static_files 
- https://github.com/zauberzeug/nicegui/blob/838421bf3090a4808a1836ab0f21029235269189/nicegui/app/app.py#L138

But for the `add_static_file` method, I do see a `Cache-Control` header from the source code 
- https://github.com/zauberzeug/nicegui/blob/838421bf3090a4808a1836ab0f21029235269189/nicegui/app/app.py#L193

**Reply by python-and-novella** [reply]
But , in stacticfiles,
```python3
if self.is_not_modified(response.headers, request_headers):
            return NotModifiedResponse(response.headers)
```
Is it useful for cache-control?

**Reply by daya0576** [reply]
There is a difference: if the `Cache-Control` header is returned with a TTL, the client can use the cache locally instead of sending a request to the server.

**Comment by rodja** [comment, marked as answer]
You are right. It seems that `fastapi.staticfiles.StaticFiles` does not add the `Cache-Control` header. But our own implementation of serving a single file does. Do you or anyone has an idea on how to add the header field to `add_static_files`?

**Reply by daya0576** [reply]
Here is a workaround that I found on the internet: https://github.com/fastapi/fastapi/discussions/7618
Not sure if it's the best solution.

BTW, same issue exists with the `_nicegui/2.3.0/static/*` files: https://github.com/zauberzeug/nicegui/blob/838421bf3090a4808a1836ab0f21029235269189/nicegui/nicegui.py#L59

**Reply by rodja** [reply]
Ah, cool. Would you like to create a pull request?

**Reply by daya0576** [reply]
Sure, I would be willing to make it during the upcoming weekend ☺️

**Reply by falkoschindler** [reply]
PR #3865 just got released in version 2.4.0 ✔️
